x-pack/plugins/fleet/server/services/agents/helpers.ts

```diff
@@ -24,15 +26,62 @@ export function searchHitToAgent(
     fields?: { status?: AgentStatus[] };
   }
 ): Agent {
-  // @ts-expect-error @elastic/elasticsearch MultiGetHit._source is optional
+  const outputs: OutputMap | undefined = hit._source?.outputs
+    ? Object.entries(hit._source?.outputs).reduce((acc, [key, val]) => {
+        acc[key] = {
+          api_key_id: val.api_key_id,
+          type: val.type,
+          to_retire_api_key_ids: val.to_retire_api_key_ids
+            ? val.to_retire_api_key_ids.map((item) => ({
+                id: item.id,
+                retired_at: item.retired_at,
+              }))
+            : undefined,
+        };
+        return acc;
+      }, {} as OutputMap)
+    : undefined;
+  const components: FleetServerAgentComponent[] | undefined = hit._source?.components
+    ? hit._source?.components.map((component) => ({
+        id: component.id,
+        type: component.type,
+        status: component.status,
+        message: component.message,
+        units: component.units.map((unit) => ({
+          id: unit.id,
+          type: unit.type,
+          status: unit.status,
+          message: unit.message,
+          // key-value pairs
+          payload: unit.payload,
+        })),
+      }))
+    : undefined;
   const agent: Agent = {
     id: hit._id,
-    ...hit._source,
+    type: hit._source?.type!,
+    active: hit._source?.active!,
+    enrolled_at: hit._source?.enrolled_at!,
+    unenrolled_at: hit._source?.unenrolled_at!,
+    unenrollment_started_at: hit._source?.unenrollment_started_at,
+    upgraded_at: hit._source?.upgraded_at,
+    upgrade_started_at: hit._source?.upgrade_started_at,
+    access_api_key_id: hit._source?.access_api_key_id,
+    default_api_key_id: hit._source?.default_api_key_id,
+    policy_id: hit._source?.policy_id,
+    last_checkin: hit._source?.last_checkin,
+    last_checkin_status: hit._source?.last_checkin_status,
+    last_checkin_message: hit._source?.last_checkin_message,
     policy_revision: hit._source?.policy_revision_idx,
-    access_api_key: undefined,
-    status: undefined,
     packages: hit._source?.packages ?? [],
     sort: hit.sort,
+    tags: hit._source?.tags,
+    outputs,
+    components,
+
+    // key-value pairs
+    user_provided_metadata: hit._source?.user_provided_metadata!,
+    local_metadata: hit._source?.local_metadata!,
```

Comment:
Left the metadata fields as is, as we don't have a defined list of fields to map. Hopefully nothing sensitive will be added here.